spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/filter/GatewayMetricsFilter.java

```diff
@@ -0,0 +1,47 @@
+package org.springframework.cloud.gateway.filter;
+
+import static org.springframework.cloud.gateway.support.ServerWebExchangeUtils.GATEWAY_ROUTE_ATTR;
+
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+
+import org.springframework.cloud.gateway.route.Route;
+import org.springframework.core.Ordered;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.server.ServerWebExchange;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tag;
+import reactor.core.publisher.Mono;
+
+public class GatewayMetricsFilter implements GlobalFilter, Ordered {
+
+	private MeterRegistry meterRegistry;
+
+	public GatewayMetricsFilter(MeterRegistry meterRegistry) {
+		this.meterRegistry = meterRegistry;
+	}
+
+	@Override
+	public int getOrder() {
+		return NettyWriteResponseFilter.WRITE_RESPONSE_FILTER_ORDER - 1;
+	}
+
+	@Override
+	public Mono<Void> filter(ServerWebExchange exchange, GatewayFilterChain chain) {
+		long start = System.nanoTime();
```

Comment:
Use `Timer.start()` instead so the timing respects alternative clocks.

Reply:
What API are you referring to? I am following the same pattern as seen in [spring metrics web fitler](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/metrics/web/reactive/server/MetricsWebFilter.java#L62)

Reply:
The Spring boot web filter preceded the availability of this API and should be updated. Check out `Timer.start()` which generates a `Sample` which you can then call `stop()` on at the end of the request.

Reply:
Done